third_party/amd/lib/TritonAMDGPUToLLVM/DotOpToLLVM/MFMA.cpp

```diff
@@ -240,13 +247,16 @@ struct DotOpMFMAConversionHelper {
           }
           acc = zeroAuxiliarBlocks(subBlocks, acc);
           for (int k = 0; k < numRepK; k++) {
-            for (int kPack = 0; kPack < kWidth / kBase; ++kPack)
+            for (int kPack = 0; kPack < kWidth / kBase; ++kPack) {
               acc =
                   mfmaLayout.getIsTransposed()
                       ? generateMFMAOp(mfmaInsnName, operandB[kPack][{b, n, k}],
                                        operandA[kPack][{b, m, k}], acc)
                       : generateMFMAOp(mfmaInsnName, operandA[kPack][{b, m, k}],
                                        operandB[kPack][{b, n, k}], acc);
+              if (b + m + n + k + kPack == 0)
```

Comment:
Use a closure before the loop (line 235) that captures this:
```
    auto getFirstMfma = [&]() {
        if (!firstMfma)
            firstMfma = acc;
    };

```

Reply:
added null check when using `firstMfma`

Reply:
If you put that lambda just after the declaration of `firstMfma`, then in the loop you can just call getFirstMfma unconditionally to capture the first one.

Reply:
Instead of the `if (b + m + n + k + kPack == 0)` logic.